File: frontend/src/types/schedule.ts
import type { AIEvidence } from './ai'

export interface Schedule {
  id: number
  customer_id: number
  user_id: number | null
  suggestion_id: number | null
  title: string
  description: string | null
  due_at: string
  priority: string
  source: string
  status: string
  outcome: string | null
  completion_note: string | null
  completed_at: string | null
  evidence: AIEvidence[]
  confirmed_by: number | null
  confirmed_at: string | null
  wecom_calendar_id: string | null
  created_at: string
  updated_at: string
}

export type ScheduleOutcome = 'contacted' | 'no_response' | 'appointment' | 'converted' | 'lost' | 'other'

export interface ScheduleCompletion {
  outcome: ScheduleOutcome
  completion_note?: string | null
}

export interface ScheduleUpdate {
  title?: string
  description?: string | null
  due_at?: string
  priority?: 'low' | 'normal' | 'high'
}

File: backend/app/schemas/schedule.py
from datetime import datetime
from typing import Any, Literal

from pydantic import BaseModel, ConfigDict, Field


class ScheduleUpdate(BaseModel):
    title: str | None = Field(default=None, min_length=1, max_length=200)
    description: str | None = Field(default=None, max_length=2000)
    due_at: datetime | None = None
    priority: str | None = Field(default=None, pattern="^(low|normal|high)$")


class ScheduleCompletion(BaseModel):
    """人工回填的跟进结果；不会自动更新客户阶段。"""

    outcome: Literal["contacted", "no_response", "appointment", "converted", "lost", "other"] = "other"
    completion_note: str | None = Field(default=None, max_length=500)


class ScheduleRead(BaseModel):
    id: int
    customer_id: int
    user_id: int | None
    suggestion_id: int | None
    title: str
    description: str | None
    due_at: datetime
    priority: str
    source: str
    status: str
    outcome: str | None
    completion_note: str | None
    completed_at: datetime | None
    evidence: list[dict[str, Any]]
    confirmed_by: int | None
    confirmed_at: datetime | None
    wecom_calendar_id: str | None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)
